cpp/models/ode_secirvvs/model.h

```diff
@@ -710,12 +711,12 @@ class Simulation : public BaseT
                         auto exceeded_threshold = dyn_npis.get_max_exceeded_threshold(inf_rel);
                         if (exceeded_threshold != dyn_npis.get_thresholds().end() &&
                             (exceeded_threshold->first > m_dynamic_npi.first ||
-                             t > double(m_dynamic_npi.second))) { //old npi was weaker or is expired
+                             t > ScalarType(m_dynamic_npi.second))) { //old npi was weaker or is expired
 
-                            auto t_start = SimulationTime(t + delay_lockdown);
+                            auto t_start = SimulationTime(t + delay_npi_implementation);
                             auto t_end   = t_start + SimulationTime(dyn_npis.get_duration());
-                            this->get_model().parameters.get_start_commuter_detection() = (double)t_start;
-                            this->get_model().parameters.get_end_commuter_detection()   = (double)t_end;
+                            this->get_model().parameters.get_start_commuter_detection() = (ScalarType)t_start;
+                            this->get_model().parameters.get_end_commuter_detection()   = (ScalarType)t_end;
```

Comment:
Since we define `double m_commuter_nondetection    = 0.0;`, we always have 100% perfect testing per default. 
That really is a problem. Good that you noticed this :)
I'm not sure how we should solve this. Actually, testing should not necessarily be coupled to the dynamic NPIs in this way. Also, we can't change the non detection value during the simulation. 
An option would be to change the default value to 1, then it's indirectly disabled.
On the other hand, it is actually a problem that currently exists and is independent of the issue. It makes the most sense to move it to its own issue, right?

Reply:
Yes, you can open a new issue and resolve this comment then

Reply:
see #1116